### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by mobileoverlord**
Update deps
* phoenix 1.4.0-rc
* ecto 3.0
* ecto_sql 3.0.0-rc
* postgrex 0.14.0-rc
* elixir 1.7.3

This also cleans up some unused alias warnings and modifies some of the code to accommodate for the ecto update.

### CODE REVIEW

File: apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 defmodule NervesHubCore.Repo.Migrations.DefaultUserOrg do
-  import Ecto.Query
+  # import Ecto.Query
```
Comment (fhunleth) [inline apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs]: Should this be deleted?

File: apps/nerves_hub_api/test/nerves_hub_api_web/controllers/user_controller_test.exs [hunk 2]
```diff
@@ -27,7 +27,7 @@ defmodule NervesHubAPIWeb.UserControllerTest do
 
   test "authenticate existing accounts" do
     password = "12345678"
-    org = Fixtures.org_fixture(%{name: "a new name"})
+    org = Fixtures.org_fixture(%{name: "a-new-name"})
```
Comment (fhunleth) [inline apps/nerves_hub_api/test/nerves_hub_api_web/controllers/user_controller_test.exs]: Did spaces in names break something or did it just offend your sensibilities?

File: apps/nerves_hub_www/test/nerves_hub_www_web/controllers/session_controller_test.exs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 defmodule NervesHubWWWWeb.SessionControllerTest do
-  use NervesHubWWWWeb.ConnCase.Browser, async: true
```
Comment (jlmcgehee21) [inline apps/nerves_hub_www/test/nerves_hub_www_web/controllers/session_controller_test.exs]: What created the need to make this synchronous?

File: apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs (hunk 1 above)
Comment (jlmcgehee21) [reply apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs]: I almost had the same question, but it's sort of documentation of what was done.  If you look at the rest of the file, you'll see that we did a data migration that required this, but you should not need to rerun the data migration every time.

File: apps/nerves_hub_www/test/nerves_hub_www_web/controllers/session_controller_test.exs (hunk 3 above)
Comment (mobileoverlord) [reply apps/nerves_hub_www/test/nerves_hub_www_web/controllers/session_controller_test.exs]: You're right, this was unnecessary. I undid it.

**Pull request merged**